sites/partners/src/components/listings/PaperListingForm/sections/ListingPhoto.tsx

```diff
@@ -149,7 +149,15 @@ const ListingPhoto = () => {
     Pass the file for the dropzone callback along to the uploader
   */
   const photoUploader = async (file: File) => {
-    void (await cloudinaryFileUploader({ file, setCloudinaryData, setProgressValue }))
+    const generatedId = await cloudinaryFileService.putFile(
+      "cloudinaryBuilding",
+      file,
+      setProgressValue
+    )
+    setCloudinaryData({
```

Comment:
Why do we need to do this in a separate step? since everything is generated from cloudinaryFileService why not just do `putFile` and `setCloudinaryData` in one go?

Reply:
setCloudinaryData is a react state function that sets the id and url fields of the CloudinaryData object, which is created and used by the client that uploads data. My reasoning here was that this id/url tuple might be specific to Cloudinary so we should keep just the id as the return value from putFile() in the generic interface and then set the values here. I did pass setProgressValue in putFile() since that is related to the progress bar and is a feature we would want to have regardless of the implementation. I'm open to passing setCloudinaryData in the function as well and having it be void instead of returning the id.

Reply:
I see got it got it. Non-blocking thought that if this is cloudinary specific we could write a function in `cloudinaryFileService` that does both  (and is not a part of fileService) but I'm not convinced myself we need to do that